pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -398,3 +406,12 @@ func (h *AutoscalersController) enqueue(obj interface{}) {
 	}
 	h.queue.AddRateLimited(key)
 }
+
+func (h *AutoscalersController) deleteFromLocalStore(toDelete []custommetrics.ExternalMetricValue) {
+	for _, d := range toDelete {
+		key := custommetrics.ExternalMetricValueKeyFunc(d)
+		h.toStore.m.Lock()
```

Comment:
What about putting the `Lock()` and `Unlock()` outside the loop since we do it for each items and the loop didn't contains any other logic.